tests/Doctrine/Tests/ORM/Functional/OidReuseTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional;
+
+
+use Doctrine\Tests\Models\CMS\CmsUser;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class OidReuseTest extends OrmFunctionalTestCase
+{
+
+    private $userId;
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function setUp()
+    {
+        $this->useModelSet('cms');
+        parent::setUp();
+    }
+
+    public function testOidReuse()
```

Comment:
Note for whoever gets to merge this: we need to first move the test to the `UnitOfWorkTest`